### PR TITLE
Add constexpr annotations to help coverity understand constant ...

### PR DESCRIPTION
**Pull request by timothy-king**
variable guards.

### CODE REVIEW

File: src/theory/quantifiers/sygus/ce_guided_single_inv_sol.cpp [hunk 1]
```diff
@@ -702,7 +702,7 @@ Node CegConjectureSingleInvSol::reconstructSolution(Node sol,
             // if it is not already reconstructed
             if( d_reconstruct.find( itt->second )==d_reconstruct.end() ){
               Trace("csi-rcons") << "...reconstructed " << ns << " for term " << nr << std::endl;
-              bool do_check = true;//getPathToRoot( itt->second );
+              constexpr bool do_check = true;//getPathToRoot( itt->second );
```
Comment (ajreynol) [inline src/theory/quantifiers/sygus/ce_guided_single_inv_sol.cpp]: I would be fine if you wanted to eliminate this variable and simplify the code accordingly.

Comment (timothy-king) [reply src/theory/quantifiers/sygus/ce_guided_single_inv_sol.cpp]: Done.

File: src/theory/bv/theory_bv.cpp [hunk 2]
```diff
@@ -861,9 +861,10 @@ bool TheoryBV::storePropagation(TNode literal, SubTheory subtheory)
   }
 
   // Propagate differs depending on the subtheory
-  // * bitblaster needs to be left alone until it's done, otherwise it doesn't know how to explain
+  // * bitblaster needs to be left alone until it's done, otherwise it doesn't
+  //   know how to explain
   // * equality engine can propagate eagerly
-  bool ok = true;
+  constexpr bool ok = true;
```
Comment (aniemetz) [inline src/theory/bv/theory_bv.cpp]: Something doesn't quite make sense here. `ok` is always true, right? So the branch on line 870 is never taken. Shouldn't line 869 be `ok = d_out->propagate(literal)`?

Comment (aniemetz) [reply src/theory/bv/theory_bv.cpp]: You can only use `constexpr` for `ok` if this is always true, right? If I address this issue in a future PR I will have to drop `constexpr` again? Maybe we should fix the underlying issue first. Is this PR urgent?

Comment (timothy-king) [reply src/theory/bv/theory_bv.cpp]: > You can only use constexpr for ok if this is always true, right?

Correct.

> If I address this issue in a future PR I will have to drop constexpr again?

This seems like a trivial cost, and if I add a comment it falls to a few seconds to resolve by deleting it and the comment.

> Maybe we should fix the underlying issue first.

I suggest going forward as it is an improvement over the current state.

> Is this PR urgent?

No.

File: src/theory/quantifiers/sygus/ce_guided_single_inv_sol.cpp [hunk 3]
```diff
@@ -704,19 +704,18 @@ Node CegConjectureSingleInvSol::reconstructSolution(Node sol,
             // if it is not already reconstructed
             if( d_reconstruct.find( itt->second )==d_reconstruct.end() ){
```
Comment (aniemetz) [inline src/theory/quantifiers/sygus/ce_guided_single_inv_sol.cpp]: Can you clang-format this whole if-block?

File: src/theory/bv/theory_bv.cpp (hunk 2 above)
Comment (aniemetz) [reply src/theory/bv/theory_bv.cpp]: I just wanted to make sure that this is what we have to do when we address this problem.
Yes, add a comment. Maybe open an issue and add the issue number. You can assign me.

**Pull request merged**